proposals/lip-0053.md

```diff
@@ -630,6 +628,13 @@ _Figure 1: A schematic of the steps of a mainchain CCU execution. Some details a
 def execute(trs: Transaction) -> None:
     ccu = trs
     ccu.params = decode(crossChainUpdateTransactionParams, trs.params)
+
+    if ccu.params.inboxUpdate is not empty:
+        # This check is expensive. Therefore, it is done in the execute step instead of the verify
+        # step. Otherwise, a malicious relayer could spam the transaction pool with computationally
+        # costly CCU verifications without paying fees.
+        verifyPartnerChainOutboxRoot(ccu)
+
     try:
         beforeCrossChainMessagesExecution(ccu)
```

Comment:
One option would be also to include lines 631-637 to `beforeCrossChainMessagesExecution`, as a first step.

Reply:
I kept now the function `verifyPartnerChainOutboxRoot` but just extracted the cheap verification steps into a new function `verifyOutboxRootWitness` which I call in the verify step.

Reply:
Looks correct to me. 

However I would still suggest to move lines 639 - 645 to `beforeCrossChainMessagesExecution`. I don't see any problem with that (perhaps I am missing something) and I think it would avoid repetition of this pseudocode and make the executions shorter and nicer.

Reply:
I see. It's indeed a bit cleaner. I moved it.